### PR TITLE
ci: add initial workflow from the starter example

### LINKED ISSUE #23: Add CI
This should:

- [ ] test
- [ ] lint/fmt
- [ ] tag/release?
- [ ] look for semver regressions?

### PR DESCRIPTION
**Pull request by jrobsonchase**
Tests probably won't pass, since there's no authtoken. I'll address that before mergetime.

Resolves #23 

Does this work?
Resolves ngrok-private/ngrok#13368

### CODE REVIEW

**Comment by jrobsonchase** [comment]
Alright, pretty happy with the test groups for the time being. We get a few basic unauthenticated tests to run at the moment, as well as any fully-offline tests.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        uses: golangci/golangci-lint-action@v3
+        with:
+          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
+          version: v1.50.0
+
+          # Optional: working directory, useful for monorepos
+          # working-directory: somedir
+
+          # Optional: golangci-lint command line arguments.
+          args: --skip-dirs=internal/muxado
```
Comment (euank) [inline .github/workflows/lint.yml:30]: I think we should remove this from `lint.yml` and put it in a `.golangci.yml` in the root with:

```yaml

run:
  skip-dirs:
  - internal/muxado
```

The intent is of course that a naive runner of just `golangci-lint` doesn't get warnings or such.

**Review by euank — COMMENTED** [review]
One suggestion, otherwise looks good!

**Review by euank — APPROVED** [review]
lgtm!

**Pull request merged**